### PR TITLE
Streamlit Altair Tutorials

### PR DESCRIPTION
**Pull request by j-gillam**
Includes:
- Getters for the data
- Altair tutorial notebook
- Streamlit tutorial python script.
- Simple streamlit app set up with altair tutorial example.

### CODE REVIEW

File: streamlit_app_tutorial/streamlit_template.py
Comment (emily-bicks) [inline streamlit_app_tutorial/streamlit_template.py:1]: Do we need the template file in the repo or was it just used as a starting point for the actual app?

File: streamlit_app_tutorial/.streamlit/secrets.toml
Comment (emily-bicks) [inline streamlit_app_tutorial/.streamlit/secrets.toml:1]: I know this is just an example, but I think best practice would be to not include the secrets file in the repo

File: streamlit_app_tutorial/README.md
Comment (emily-bicks) [inline streamlit_app_tutorial/README.md:1]: A few (super minor) suggestions:
- may be worth prefacing the setup with something like "run set up to get the app running on your local machine", just so it's clear you don't need to do this to just view the app?
- As mentioned in my previous comment, I think best practice is not to include the secrets.toml file in the repo, but instead instruct users in the readme on where to put it and the format
- on bullet 6, add `.py`
- not following why we would need to change the path in the code to have someone run it locally? I imagine this has something to do with the directory structure needed for streamlit cloud, but could the local structure not match that?
- would link to where you got the data in the "About the Data" section

File: streamlit_app_tutorial/streamlit_demo_iod_analysis.py [hunk 1]
```diff
@@ -12,23 +12,29 @@
 from getters.lsoa_shapefiles_2011 import get_english_lsoa_shapefiles_2011
 import os
 
+# Make sure to read the README.md file for this app before running this code.
+
 
 # Caches the data to prevent computation on every rerun
 @st.cache_data
```
Comment (emily-bicks) [inline streamlit_app_tutorial/streamlit_demo_iod_analysis.py]: You can do @st.cache_data(show_spinner = "Loading data") to specify the message that the user sees while the app is spinning up

I can't comment directly on this line, but on line 165, I would put the steps like this that manipulate the data inside a function that's cached

File: streamlit_app_tutorial/streamlit_demo_iod_analysis.py
Comment (emily-bicks) [inline streamlit_app_tutorial/streamlit_demo_iod_analysis.py:1]: On the Home page I would add something like "You must input the password (Nest@Iod) to access this app, if you don't input the password, the tabs will show up as blank" - I think the password is just for illustrative purposes to demo that feature right? so I would just put what it is right on the home page.

File: streamlit_app_tutorial/getters/english_la_iod_data_2019.py
Comment (emily-bicks) [inline streamlit_app_tutorial/getters/english_la_iod_data_2019.py:1]: Only comment for the getters it that I would add links to where the data comes from in the docstring

**Review by emily-bicks — APPROVED** [review]
Looks really great - awesome app and super well commented!
Just made a few minor suggestions, but I think the main things are:
- if this is in a medium article, I assume the password is just for illustrative purposes, so I would just write that and what it is on the homepage so people realize (I didn't at first and I was wondering why all the tabs were blank)
- In the readme, I would add a bit about why you may want to run the setup process (i.e. to edit the code or run it locally), but that it's not needed to play around with the front end
- I would link to the actual documentation / websites where you downloaded the datasets in the readme, getters, and notebook when you load the data

All just minor suggestions, but again looks great!!

File: streamlit_app_tutorial/streamlit_template.py
Comment (j-gillam) [reply streamlit_app_tutorial/streamlit_template.py:1]: I wanted to give a less complicated example for people to look at the code for.

File: streamlit_app_tutorial/streamlit_demo_iod_analysis.py
Comment (j-gillam) [reply streamlit_app_tutorial/streamlit_demo_iod_analysis.py:1]: Yep just illustrative!

**Pull request merged**